composer/core/state.py

```diff
@@ -170,7 +167,6 @@ def __init__(
         rank_zero_seed: int,
 
         # data configurations
-        train_dataloader: types.DataLoader,
```

Comment:
I did not catch the subtlety in the design documentation, but prefer to keep the dataloader in the `__init__`, this is consistent with other trainers.

Reply:
Leaving the dataloader in init for now.